packages/react-core/src/components/Alert/__tests__/AlertToggleExpandButton.test.tsx

```diff
@@ -0,0 +1,150 @@
+import * as React from 'react';
+import { render, screen } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+
+import { AlertToggleExpandButton } from '../AlertToggleExpandButton';
+import { AlertContext } from '../AlertContext';
+
+jest.mock('../../Button', () => ({
+  Button: ({ children, variant, isInline, onClick, ...props }) => (
+    <>
+      <button onClick={onClick} {...props}>
+        {children}
+      </button>
+      <p>{`variant: ${variant}`}</p>
+      <p id="labelling-id">Test label</p>
+    </>
+  ),
+  ButtonVariant: { plain: 'plain' }
+}));
+
+jest.mock('@patternfly/react-icons/dist/esm/icons/angle-right-icon', () => () => 'Icon mock');
+
+test('Renders without children', () => {
+  render(
+    <div data-testid="container">
+      <AlertContext.Provider value={{ title: 'title', variantLabel: 'variantLabel' }}>
+        <AlertToggleExpandButton />
+      </AlertContext.Provider>
+    </div>
+  );
+
+  expect(screen.getByTestId('container').firstChild).toBeVisible();
+});
+
+test('Renders with an aria label composed with the title and variantLabel provided via a context by default', () => {
+  render(
+    <AlertContext.Provider value={{ title: 'title', variantLabel: 'variantLabel' }}>
+      <AlertToggleExpandButton />
+    </AlertContext.Provider>
+  );
+
+  expect(screen.getByRole('button')).toHaveAccessibleName('Toggle variantLabel alert: title');
+});
+
+test('Renders with the aria label provided via prop when one is provided', () => {
+  render(
+    <AlertContext.Provider value={{ title: 'title', variantLabel: 'variantLabel' }}>
+      <AlertToggleExpandButton aria-label="Aria label prop" />
+    </AlertContext.Provider>
+  );
+
+  expect(screen.getByRole('button')).toHaveAccessibleName('Aria label prop');
+});
+
+test('Does not call the callback provided via onClose when it is not clicked', () => {
+  const onCloseMock = jest.fn();
+
+  render(
+    <AlertContext.Provider value={{ title: 'title', variantLabel: 'variantLabel' }}>
+      <AlertToggleExpandButton onToggleExpand={onCloseMock} />
+    </AlertContext.Provider>
+  );
+
+  expect(onCloseMock).not.toHaveBeenCalled();
+});
```

Comment:
Nit: could we update this test to reflect the onToggleExpand is being tested instead of onClose?

Reply:
Yes! Great catch!